### PR TITLE
Vue: Add first-class args support

### PR DESCRIPTION
**Pull request by shilman**
Issue: #11051 

## What I did

Worked  with @backbone87 and @tmeasday  to automatically inject args into the prop values, so you can do this:

```js
export const Rounded = (args) => ({
  props: Object.keys(args),
  components: { MyButton },
  template: '<my-button :color="color" :rounded="rounded">A Button with rounded edges</my-button>',
});
Rounded.argTypes = {
  rounded: { defaultValue: true },
  color: { control: { type: 'color' }, defaultValue: '#f00' },
};
```

This needs documentation but I'm not sure where is the best place for it currently.

## How to test

See `vue-kitchen-sink`

Feedback wanted, h/t @ndelangen for the idea of baking it into `@storybook/vue`

### CODE REVIEW

File: app/vue/src/client/preview/render.ts [hunk 1]
```diff
@@ -44,8 +44,10 @@ export default function render({
 
   showMain();
 
-  // always refresh the component
-  root[COMPONENT] = element;
+  // at component creation || refresh by HMR	  // always refresh the component
```
Comment (tmeasday) [inline app/vue/src/client/preview/render.ts:47]: Refresh by HMR OR changing story to be clear.

**Comment by graup** [comment]
> If the story returns args, @storybook/vue will unpack the args into props with default values.

To me, that doesn't sound very intuitive. If we introduce "magic", it should be properly named, something like `{ defaultProps: args }`. But is it really necessary? It would be better if we can let the user define data/props normally and just work with that.

**Comment by shilman** [comment]
@graup point taken. check out the new version, updated in the description. much better i think!

**Comment by shilman** [comment]
Self-merging @backbone87 @tmeasday

**Comment by graup** [comment]
@shilman yeah this is better! 👍

**Comment by Martskin** [comment]
The controls are working on the Canvas tab great now but don't see to be having any effect in the Docs addon tab.

https://next--storybookjs.netlify.app/vue-kitchen-sink/?path=/story/button--rounded

storybook-doc-controls

**Comment by shilman** [comment]
@Martskin  thanks for letting me know. Will look into it tomorrow!

**Comment by Martskin** [comment]
> @Martskin thanks for letting me know. Will look into it tomorrow!

Glad to help! Thanks for the repsone.

**Comment by Martskin** [comment]
> The controls are working on the Canvas tab great now but don't seem to be having any effect in the Docs addon tab.

Release https://github.com/storybookjs/storybook/releases/tag/v6.0.0-beta.26 fixed it for me! Thanks.

**Pull request merged**